File: app/src/main/java/era/apps/happinessjar/models/slider/adapter/CustomPagerAdapter.java
package era.apps.happinessjar.models.slider.adapter;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;

import era.apps.happinessjar.R;
import era.apps.happinessjar.util.callback.OnItemClick;


public class CustomPagerAdapter extends PagerAdapter {
    private int[] modelList;


    public void setModelList(int[] modelList) {
        this.modelList = modelList;
        notifyDataSetChanged();
    }


    @Override
    public int getCount() {
        return modelList.length;
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object o) {
        return view == o;
    }

    private OnItemClick onItemClick;

    public void setOnItemClick(OnItemClick onItemClick) {
        this.onItemClick = onItemClick;
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup collection, int position) {

        LayoutInflater inflater = LayoutInflater.from(collection.getContext());
        View view = inflater.inflate(modelList[position], collection, false);
        if (position == modelList.length - 1) {
            view.findViewById(R.id.finishSlider).setOnClickListener(view1 -> onItemClick.OnClick(0));
        }
        collection.addView(view);
        return view;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup parant, int position, @NonNull Object view) {
        parant.removeView((View) view);
    }

}

File: app/src/main/java/era/apps/happinessjar/ui/adapter/CustomPagerAdapter.kt
package era.apps.happinessjar.ui.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
import era.apps.happinessjar.R
import era.apps.happinessjar.util.callback.OnItemClick


class CustomPagerAdapter : PagerAdapter() {
    private lateinit var modelList: IntArray
    fun setModelList(modelList: IntArray) {
        this.modelList = modelList
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return modelList.size
    }

    override fun isViewFromObject(view: View, o: Any): Boolean {
        return view === o
    }

    private var onItemClick: OnItemClick? = null
    fun setOnItemClick(onItemClick: OnItemClick?) {
        this.onItemClick = onItemClick
    }

    override fun instantiateItem(collection: ViewGroup, position: Int): Any {
        val inflater = LayoutInflater.from(collection.context)
        val view = inflater.inflate(modelList[position], collection, false)
        if (position == modelList.size - 1) {
            view.findViewById<View>(R.id.finishSlider).setOnClickListener {
            onItemClick!!.onClick(0) }
        }
        collection.addView(view)
        return view
    }

    override fun destroyItem(parant: ViewGroup, position: Int, view: Any) {
        parant.removeView(view as View)
    }
}
